Obfuscar/MethodKey.cs

```diff
@@ -180,10 +180,17 @@ static string ToGenerelGenericMemberName(string memberName)
                         break;
                     case '>':
                         level--;
+                        
+                        //if still within generic or nested generic type listings
+                        if (level >= 0)
```

Comment:
Could you add some tests for this?

Reply:
Sure but lets talk about which level of testing we want here as none of the ___key classes are tested directly but used indirectly in many other tests.